src/Number.php

```diff
@@ -20,17 +20,37 @@
     private $fractionalPart;
 
     /**
+     * @var array
+     */
+    private static $numbers = [0 => 1, 1 => 1, 2 => 1, 3 => 1, 4 => 1, 5 => 1, 6 => 1, 7 => 1, 8 => 1, 9 => 1];
+
+    /**
+     * @var array
+     */
+    private static $signs = ['-' => 1];
+
+    /**
      * @param string $integerPart
      * @param string $fractionalPart
      */
     public function __construct($integerPart, $fractionalPart = '')
     {
-        if ($this->validateNumberAsInteger($integerPart) === false) {
-            throw new \InvalidArgumentException('Invalid number');
+        if ($integerPart !== '' && $this->validateAsIntegerPart($integerPart) === false) {
+            throw new \InvalidArgumentException(
+                'Invalid number, integer part '.$integerPart.' is not an integer'
+            );
         }
 
-        if ($fractionalPart !== '' && $this->validateNumberAsInteger($fractionalPart) === false) {
-            throw new \InvalidArgumentException('Invalid number');
+        if ($fractionalPart !== '' && $this->validateAsFractionalPart($fractionalPart) === false) {
+            throw new \InvalidArgumentException(
+                'Invalid number, fractional part '.$integerPart.' is not an integer'
+            );
+        }
+
+        if ($integerPart === '-') {
+            $integerPart = '-0';
+        } else {
+            $integerPart = ltrim($integerPart, '0');
```

Comment:
We should decide now what to do with it. Afterwards, it becomes a feature (or not). I have nothing against, but I also see no reason why to support it.